In [ ]:
import numpy as np
import pandas as pd
import matplotlib.pyplot as plp

from sklearn.preprocessing import PolynomialFeatures, OneHotEncoder, LabelEncoder, StandardScaler
from sklearn.model_selection import train_test_split
from sklearn.ensemble import RandomForestClassifier
from sklearn.metrics import accuracy_score, confusion_matrix


In [2]:
df = pd.read_csv('heart_failure_clinical_records_dataset.csv')

In [4]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 299 entries, 0 to 298
Data columns (total 13 columns):
 #   Column                    Non-Null Count  Dtype  
---  ------                    --------------  -----  
 0   age                       299 non-null    float64
 1   anaemia                   299 non-null    int64  
 2   creatinine_phosphokinase  299 non-null    int64  
 3   diabetes                  299 non-null    int64  
 4   ejection_fraction         299 non-null    int64  
 5   high_blood_pressure       299 non-null    int64  
 6   platelets                 299 non-null    float64
 7   serum_creatinine          299 non-null    float64
 8   serum_sodium              299 non-null    int64  
 9   sex                       299 non-null    int64  
 10  smoking                   299 non-null    int64  
 11  time                      299 non-null    int64  
 12  DEATH_EVENT               299 non-null    int64  
dtypes: float64(3), int64(10)
memory usage: 30.5 KB


In [11]:
x = df.drop('DEATH_EVENT', axis=1)
y = df['DEATH_EVENT']

In [12]:
x_train, x_test, y_train, y_test = train_test_split(x, y, test_size=0.2, random_state=42, stratify=y)

In [13]:
model = RandomForestClassifier(random_state=42)

model.fit(x_train, y_train)

y_pred = model.predict(x_test)
print(f'Accuracy score: {accuracy_score(y_test, y_pred)}')
print(confusion_matrix(y_test, y_pred))

Accuracy score: 0.8333333333333334
[[38  3]
 [ 7 12]]


In [5]:

poly = PolynomialFeatures(degree=2, include_bias=False)

x_poly = poly.fit_transform(df)

poly_column = poly.get_feature_names_out(input_features=df.columns)
df_poly = pd.DataFrame(x_poly, columns=poly_column)

In [6]:
df_poly.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 299 entries, 0 to 298
Columns: 104 entries, age to DEATH_EVENT^2
dtypes: float64(104)
memory usage: 243.1 KB


In [7]:
df_poly.shape

(299, 104)

In [8]:
x = df_poly.drop('DEATH_EVENT', axis=1)
y = df_poly['DEATH_EVENT']

In [9]:
x_train, x_test, y_train, y_test = train_test_split(x, y, test_size=0.2, random_state=42, stratify=y)

In [10]:
model = RandomForestClassifier(random_state=42)

model.fit(x_train, y_train)

y_pred = model.predict(x_test)
print(f'Accuracy score: {accuracy_score(y_test, y_pred)}')
print(confusion_matrix(y_test, y_pred))

Accuracy score: 1.0
[[41  0]
 [ 0 19]]


Model performance is 
Accuracy score: 0.8333333333333334
[38  3]
[ 7 12]

After polPolynomialfeaturesy transformation performance increased.
Accuracy score: 1.0
[41  0]
[ 0 19]